source/agora/consensus/EnrollmentManager.d

```diff
@@ -241,12 +241,16 @@ public class EnrollmentManager
         assumeSafeAppend(enrolls);
 
         static Enrollment[] pool_enrolls;
-        this.enroll_pool.getEnrollments(pool_enrolls);
+        this.enroll_pool.getEnrollments(pool_enrolls, Height(height + 1));
         foreach (enroll; pool_enrolls)
         {
             const enroll_height = this.getEnrolledHeight(enroll.utxo_key);
+            const avail_height =
+                this.enroll_pool.getAvailableHeight(enroll.utxo_key);
+            assert(avail_height != Height(0));
             if (enroll_height == ulong.max ||
-                height >= enroll_height + this.params.ValidatorCycle - 1)
+                (avail_height >= enroll_height &&
+                    height >= enroll_height + this.params.ValidatorCycle - 1))
```

Comment:
The person who enrolls the information will feel bored because the information he or she has enrolled is not contained in the block and there is no change.

Reply:
The code is for the node which is willing to re-enroll as a validator although it is a validator. The code means as follows in fact.
`(enrolled_height != ulong.max && avail_height >= enroll_height && height >= enroll_height + this.params.ValidatorCycle - 1))`
If the `enrolled_height` equals to `ulong.max`, it will be contained in a block probably because it is a candidate for a validator.
Thanks.